In [23]:
import statsmodels.api as sm
import pandas as pd

In [5]:
df_boston = pd.read_csv('Boston House Prices.csv')

# Rooms: nº de habitaciones de la casa
# Distance: distancia media de la casa a 5 centros de empleo en Boston
# Value: valor medio en miles de $ 

In [6]:
df_boston

Rooms  Distance  Value
0    6.575    4.0900   24.0
1    6.421    4.9671   21.6
2    7.185    4.9671   34.7
3    6.998    6.0622   33.4
4    7.147    6.0622   36.2
..     ...       ...    ...
501  6.593    2.4786   22.4
502  6.120    2.2875   20.6
503  6.976    2.1675   23.9
504  6.794    2.3889   22.0
505  6.030    2.5050   11.9

[506 rows x 3 columns]

# 3. Regresión Lineal Múltiple con Statsmodels

## 3.1 Definir las Variables Dependiente e Independientes

In [9]:
Y = df_boston['Value'] # variable dependiente
X = df_boston[['Rooms', 'Distance']] # variables independientes

## 3.2 Crear la regresión: Agregar constante y ajustar modelo

In [11]:
X = sm.add_constant(X) # agregar una constante
lm = sm.OLS(Y, X).fit() # ajustar el modelo

In [13]:
lm.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Value   R-squared:                       0.496
Model:                            OLS   Adj. R-squared:                  0.494
Method:                 Least Squares   F-statistic:                     247.0
Date:                Thu, 27 Apr 2023   Prob (F-statistic):           1.84e-75
Time:                        13:40:23   Log-Likelihood:                -1667.1
No. Observations:                 506   AIC:                             3340.
Df Residuals:                     503   BIC:                             3353.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        -34.6361      2.621    -13.212      0.000     -39.786     -29.486
Rooms          8.8014      0.424     20.780      0.000       7.969       9.634
Distance       0.4888      0.141      3.459      0.001       0.211       0.767
==============================================================================
Omnibus:                      142.807   Durbin-Watson:                   0.684
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              844.520
Skew:                           1.090   Prob(JB):                    4.12e-184
Kurtosis:                       8.942   Cond. No.                         68.7
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

- Observamos que nuestro R-squared (R^2) es 0.496, por tanto, el nº de habitaciones y la distancia explica el 49.6% de la variabilidad de los precios de las casas.
